x-pack/plugins/fleet/common/types/rest_spec/agent.ts

```diff
@@ -100,13 +102,15 @@ export interface PostBulkAgentUpgradeRequest {
   };
 }
 
-export type PostBulkAgentUpgradeResponse = Record<
-  Agent['id'],
-  {
-    success: boolean;
-    error?: string;
-  }
->;
+export type PostBulkAgentUpgradeResponse =
```

Comment:
nit: Since this type is repeated several times on this file, it could make sense to create a type for it and reassign wherever is used? Something like this:
```
  type AgentUpgradeResponse = | Record<
      Agent['id'],
      {
        success: boolean;
        error?: string;
      }
    >
  | { actionId: string };
```
In the case you like the suggestion feel free to pick a better name!

Reply:
yes, I just did that!